modules/tests/step.rs

```diff
@@ -0,0 +1,49 @@
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Step {
+    pub action: Action,
+
+    #[serde(alias = "actionOutcome")]
+    pub action_outcome: ActionOutcome,
+
+    pub chains: HashMap<String, Chain>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Action {
+    #[serde(alias = "type")]
+    pub action_type: ActionType,
+
+    #[serde(alias = "chainId")]
+    pub chain_id: Option<String>,
+
+    #[serde(alias = "clientId")]
+    pub client_id: Option<u64>,
+
+    #[serde(alias = "clientHeight")]
+    pub client_height: Option<u64>,
+}
+
```

Comment:
For create client we need to have a `client_state` and for update client we need a `header`. It is true that for the mock client these boil down to a `client_height` but if we want to extend these tests for other client types or make changes to the mock client (e.g. add a `chain-id` or client specific parameters to the mock state, or add a timestamp to the header/ consensus state)

Reply:
Currently the model does not distinguish between client state or consensus state. For each client, only the current height is kept. For this reason, there's no `"ICS02ConsensusStateNotFound"` error in the model, only `"ICS02ClientNotFound"` (as they cannot be distinguished).

Are you suggesting that the model should be extended to make the distinction? In general, we cannot hope to model everything, but maybe this is one of the situations where we should. At this point it is not obvious to me why the current abstraction is insufficient. It might become clear once there are other client types to be tested.

Reply:
> Are you suggesting that the model should be extended to make the distinction? 

Yes

> In general, we cannot hope to model everything, but maybe this is one of the situations where we should. At this point it is not obvious to me why the current abstraction is insufficient. It might become clear once there are other client types to be tested.

Understood we cannot model everything. But the create_client message includes a client state (`AnyClientState`) and a consensus state (`AnyConsensusState`) and I think we should pass those in the `action`.  update_client should take `AnyHeader`. 
It's not only for the different client types and MBT testing for clients. When we do MBT for connection we will need to pass `AnyConsensusState` (and not a client height) in some of the messages. And this would be something that we should have previously created a client with (or not, depending on the test).

Reply:
> Understood we cannot model everything. But the create_client message includes a client state (`AnyClientState`) and a consensus state (`AnyConsensusState`) and I think we should pass those in the `action`. update_client should take `AnyHeader`.

I can make this distinction in the model, but it's not clear to me what it brings to ICS02. For example, what should be a client state and a consensus state in the model? I can model both as a height, but it feels like this is the same as having a single height (which is what's modeled now).

> It's not only for the different client types and MBT testing for clients. When we do MBT for connection we will need to pass `AnyConsensusState` (and not a client height) in some of the messages. And this would be something that we should have previously created a client with (or not, depending on the test).

If the distinction will be needed for ICS03, can we address this concern in the PR with MBT for ICS03?

Reply:
We don't have to hold this PR necessarily.  I will leave it up to you. But I think we need to make the change so that we can easily extend the tests. And just to clarify this is not only needed for ICS03.

Reply:
After discussing with @ancazamfir, we have reached the conclusion that the model should keep, per client, the latest `N` heights, not just the latest one. The current model is a special case where `N = 1`. I've opened an issue to track this: https://github.com/informalsystems/ibc-rs/issues/633. Thanks @ancazamfir!